packages/dataviews/src/components/dataviews-filters/search-widget.tsx

```diff
@@ -322,6 +335,27 @@ function ComboboxList( { view, filter, onChangeView }: SearchWidgetProps ) {
 }
 
 export default function SearchWidget( props: SearchWidgetProps ) {
-	const Widget = props.filter.elements.length > 10 ? ComboboxList : ListBox;
-	return <Widget { ...props } />;
+	const { elements, isLoading } = useElements(
+		props.filter.elements,
+		props.filter.getElements
+	);
+
+	if ( isLoading ) {
+		return (
+			<div className="dataviews-filters__search-widget-loading">
+				<Spinner />
+			</div>
+		);
+	}
+
+	if ( elements.length === 0 ) {
+		return (
+			<div className="dataviews-filters__search-widget-loading">
```

Comment:
In this case we are not loading, should the loading class be used here anyway?

Reply:
Renamed it for clarity at https://github.com/WordPress/gutenberg/pull/72254/commits/72a29a0ddfd5106d8872207f0c7ff0bef168cddf